### PR TITLE
NAS-124088 / 24.04 / Fix dumping `List` that is `None`

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 0:10

**Comment by yocalebo** [comment]
Should we add unit tests to test this specific scenario?

File: src/middlewared/middlewared/pytest/unit/test_schema.py [hunk 1]
```diff
@@ -496,6 +496,17 @@ def listnotnull(self, data):
...
 
+def test__schema_list_null_items():
+
+    @accepts(List('data', null=True, items=[Str('item')]))
+    def listnull(self, data):
+        return data
+
+    self = Mock()
+
+    listnull(self, None)
```
Comment (yocalebo) [inline src/middlewared/middlewared/pytest/unit/test_schema.py:507]: Sorry, I must me misunderstanding but how does this test anything if we're not checking return value of `listnull`?

Comment (yocalebo) [reply src/middlewared/middlewared/pytest/unit/test_schema.py:507]: Ever other test in this file checks return values and/or if an error is raised. Either:
1. add a comment on why this test is different
2. or actually do `listnull(self, None) == []` and add a `TODO:` comment so we can come back to this and fix it later

**Review by yocalebo — APPROVED** [review]
Thank you!

**Pull request merged**